### PR TITLE
Mepts 189 quality improvement report 2019

### PR DESCRIPTION
**Pull request by cfaife**

### CODE REVIEW

**Comment by ningosi** [comment]
Resolve the conflicts to this PR

**Comment by ynurmahomed** [comment]
Hi @cfaife base branch for the PR should be `develop` and not `master`. I've changed it and now there are some conflicts.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovementCohortQueries.java [hunk 1]
```diff
@@ -0,0 +1,2405 @@
...
+        EptsReportUtils.map(
+            getPacientsStartedARVInAPeriodAndHadEncounter33DaysAfterBegining(), mappings));
+
+    cd.setCompositionString("AMOSTRAGRAVIDA AND SEGUNDAVISITA");
+    return cd;
+  }
+
+  /**
+   * MQ_GRAVIDAS INSCRITAS NO SERVICO TARV (AMOSTRA GRAVIDA) E QUE TIVERAM 3 CONSULTAS OU 3
+   * LEVANTAMENTOS NOS PRIMEIROS 3 MESES DE TARV - NOVO Sao grávida
```
[the remark this answers is not in the document]
Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovementCohortQueries.java]: done

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupQualityImprovementReport.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    reportDefinition.setName(getName());
+    reportDefinition.setDescription(getDescription());
+    reportDefinition.setParameters(getParameters());
+    reportDefinition.addDataSetDefinition(
+        "ALL", Mapped.mapStraightThrough(initQltyImpDataSet.constructInitQltyImpDataSet()));
+
+    // add a base cohort here to help in calculations running
+    reportDefinition.setBaseCohortDefinition(
+        EptsReportUtils.map(
+            qualityImprovementCohortQueries.getBaseCohort(),
```
[the remark this answers is not in the document]
Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupQualityImprovementReport.java]: done

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovementCohortQueries.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovementCohortQueries.java]: It helps to  track  the cohort when  comparing with FGH report

File: omod/.externalToolBuilders/Maven_Ant_Builder.launch [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (cfaife) [reply omod/.externalToolBuilders/Maven_Ant_Builder.launch]: done, added to  git ignore file

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovementCohortQueries.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovementCohortQueries.java]: In the future they might be  needed

**Pull request merged**